pymeasure/instruments/lakeshore/lakeshore421.py

```diff
@@ -0,0 +1,395 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2019 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, \
+    truncated_discrete_set
+
+from pyvisa import constants as visa_const
+import numpy as np
+from time import sleep
+
+
+class LakeShore421(Instrument):
+    """
+    Represents the Lake Shore 421 Gaussmeter and provides
+    a high-level interface for interacting with the instrument.
+    .. code-block:: python
+
+        gaussmeter = LakeShore421("COM1")
+        gaussmeter.unit = "T"               # Set units to Tesla
+        gaussmeter.auto_range = True        # Turn on auto-range
+        gaussmeter.fast_mode = True         # Turn on fast-mode
+
+    """
+
+    MULTIPLIERS = {1e3: 'k', 1: '', 1e-3: 'm', 1e-6: 'n'}
+    RANGES = {0.003: 0, 0.03: 1, 0.3: 2, 3: 3}  # in Tesla
+    PROBETYPES = {"High Sensitivity": 0,
+                  "High Stability": 1,
+                  "Ultra-High Sensitivity": 2}
+
+    def __init__(self, resource_name, **kwargs):
+        super(LakeShore421, self).__init__(
+            resource_name,
+            "Lake Shore 421 Gaussmeter",
+            read_termination='\r',
+            write_termination='\n',
+            command_delay=0.05,
+            **kwargs
+        )
+
+        self.adapter.connection.set_visa_attribute(
+            visa_const.VI_ATTR_ASRL_BAUD, 9600)
+        self.adapter.connection.set_visa_attribute(
```

Comment:
I wonder: why are you using the `set_visa_attribute` method, instead of the [simpler](https://pyvisa.readthedocs.io/en/latest/api/resources.html?highlight=delay#pyvisa.resources.SerialInstrument) `connection.baud_rate`, `connection.data_bits` etc. attributes? There'd also be no need to drag in visa_const, then.

Reply:
This is a remainder from way back when this instrument was last used by me (I then didn't finish the implementation); this was needed for some reason then and I didn't think too much about that part when finishing it; I'll change this into the simpler attributes.

Reply:
In the end, when we resolve the safe_keywords issue, these can be passed as kwargs to the adapter constructor.